chainer/distributions/multivariate_normal.py

```diff
@@ -0,0 +1,136 @@
+import chainer
+from chainer.backends import cuda
+from chainer import distribution
+from chainer.functions.array import broadcast
+from chainer.functions.array import expand_dims
+from chainer.functions.array import repeat
+from chainer.functions.array import rollaxis
+from chainer.functions.array import squeeze
+from chainer.functions.array import swapaxes
+from chainer.functions.math import basic_math
+from chainer.functions.math import exponential
+from chainer.functions.math import inv
+from chainer.functions.math import matmul
+from chainer.functions.math import sum
+import numpy
+import math
+
+ENTROPYC = 0.5 * math.log(2 * math.pi * math.e)
+LOGPROBC = - 0.5 * math.log(2 * math.pi)
+
+
+class MultivariateNormal(distribution.Distribution):
+
+    """MultivariateNormal Distribution.
+    Args:
+        loc(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`): Parameter of distribution representing the \
+        location :math:`\\mu`.
+        scale_tril(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`): Parameter of distribution representing the \
+        scale :math:`L`.
+    """
+
+    def __init__(self, loc, scale_tril):
+        if isinstance(loc, chainer.Variable):
+            self.loc = loc
+        else:
+            self.loc = chainer.Variable(loc)
+        if isinstance(scale_tril, chainer.Variable):
+            self.scale_tril = scale_tril
+        else:
+            self.scale_tril = chainer.Variable(scale_tril)
+        self.d = self.scale_tril.shape[-1]
+
+    def __copy__(self):
+        return self._copy_to(MultivariateNormal(self.loc, self.scale_tril))
+
+    @property
+    def batch_shape(self):
+        return self.loc.shape[:-1]
+
+    @property
+    def entropy(self):
+        st = rollaxis.rollaxis(self.scale_tril, -2, 0)
+        st = rollaxis.rollaxis(st, -1, 1)
+        diag = st[list(range(self.d)), list(range(self.d))]
+        return sum.sum(exponential.log(basic_math.absolute(diag)), axis=0) \
+            + ENTROPYC * self.d
+
+    @property
+    def event_shape(self):
+        return self.loc.shape[-1:]
+
+    @property
+    def _is_gpu(self):
+        return isinstance(self.loc.data, cuda.ndarray)
+
+    def log_prob(self, x):
+        st = rollaxis.rollaxis(self.scale_tril, -2, 0)
+        st = rollaxis.rollaxis(st, -1, 1)
+        diag = st[list(range(self.d)), list(range(self.d))]
+        logdet = sum.sum(exponential.log(basic_math.absolute(diag)), axis=0)
```

Comment:
Why don't you add `_logdet` method, which would be called by `entropy` and `log_prob`?